packages/commands/src/query/cctx.ts

```diff
@@ -0,0 +1,253 @@
+import axios from "axios";
+import { Command } from "commander";
+import { ethers } from "ethers";
+import EventEmitter from "eventemitter3";
+import { z } from "zod";
+
+import { CrossChainTx } from "../../../../types/cctx";
+
+/**
+ * Event map:
+ *  - `cctx` → emitted every polling round with the **entire** array so far.
+ */
+interface CctxEvents {
+  cctx: (allSoFar: CrossChainTx[]) => void;
+}
+
+export const cctxEmitter = new EventEmitter<CctxEvents>();
+
+const cctxOptionsSchema = z.object({
+  delay: z.coerce.number().int().positive().default(2000),
+  hash: z.string(),
+  rpc: z.string(),
+});
+
+type CctxOptions = z.infer<typeof cctxOptionsSchema>;
+
+interface CctxResponse {
+  CrossChainTxs: CrossChainTx[];
+}
+
+const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+const fetchCctx = async (
+  hash: string,
+  rpc: string
+): Promise<CrossChainTx[]> => {
+  const url = `${rpc}/zeta-chain/crosschain/inboundHashToCctxData/${hash}`;
+  const res = await axios.get<CctxResponse>(url, {
+    validateStatus: (s) => s === 200 || s === 404,
+  });
+  return res.status === 404 ? [] : res.data.CrossChainTxs;
+};
```

Comment:
This could benefit from a timeout config.

```suggestion
const fetchCctx = async (
  hash: string,
  rpc: string,
  timeoutMs = 10000
): Promise<CrossChainTx[]> => {
  const url = `${rpc}/zeta-chain/crosschain/inboundHashToCctxData/${hash}`;
  const res = await axios.get<CctxResponse>(url, {
    timeout: timeoutMs,
    validateStatus: (s) => s === 200 || s === 404,
  });
  return res.status === 404 ? [] : res.data.CrossChainTxs;
};
```